NightscoutUploadKit/NightscoutUploader.swift

```diff
@@ -169,7 +169,7 @@ public class NightscoutUploader: NSObject {
                 "timestamp": pumpDateStr,
                 "bolusiob": status.iob,
             ],
-            "reservoir": status.reservoirRemainingUnits,
+            "reservoir": Double(round(status.reservoirRemainingUnits*1000)/1000,
```

Comment:
I was a bit suspicious of this solution, too, but I did a test, and it appears to work. I think it's actually a rounding error from earlier math, rather than a serialization issue, so rounding here works. If this fails, it's easy enough to back out.